test/test_rpc.py

```diff
@@ -1980,38 +1987,45 @@ def consumer_removed():
 class TestSSL(object):
 
     @pytest.fixture(params=[True, False])
-    def rabbit_ssl_config(self, request, rabbit_ssl_config):
+    def rabbit_ssl_options(self, request, rabbit_ssl_options):
         verify_certs = request.param
         if verify_certs is False:
             # remove certificate paths from config
-            rabbit_ssl_config['AMQP_SSL'] = True
-        return rabbit_ssl_config
+            options = True
+        else:
+            options = rabbit_ssl_options
+        return options
+
+    @pytest.mark.usefixtures("rabbit_ssl_config")
+    def test_rpc_entrypoint_over_ssl(self, container_factory, rabbit_uri):
 
-    def test_rpc_entrypoint_over_ssl(
-        self, container_factory, rabbit_ssl_config, rabbit_config
-    ):
         class Service(object):
             name = "service"
 
             @rpc
             def echo(self, *args, **kwargs):
                 return args, kwargs
 
-        container = container_factory(Service, rabbit_ssl_config)
+        container = container_factory(Service)
         container.start()
 
-        with ServiceRpcClient("service", rabbit_config) as client:
+        with ServiceRpcClient("service", uri=rabbit_uri, ssl=None) as client:
```

Comment:
This `ssl=None` isn't necessary. Was it added for readability?

Reply:
Yep, it is there on purpose <tl;dr> sets up SSL service, but NONSSL client.

This test uses `rabbit_ssl_config` fixture so the SSL is loaded from config where it is present. With the args I'm telling the client don't use the SSL setup from config. The *service* is run with SSL config, but the *client* is set NOT to use any SSL, similarly the client has the SSL rabbit url (which it would also load from config) is overriden with non SSL uri.